nano/node/node.hpp

```diff
@@ -214,31 +214,24 @@ class block_arrival
 	static size_t constexpr arrival_size_min = 8 * 1024;
 	static std::chrono::seconds constexpr arrival_time_min = std::chrono::seconds (300);
 };
-class rep_last_heard_info
-{
-public:
-	std::chrono::steady_clock::time_point last_heard;
-	nano::account representative;
-};
 class online_reps
 {
 public:
-	online_reps (nano::node &);
-	void vote (std::shared_ptr<nano::vote> const &);
-	void recalculate_stake ();
+	online_reps (nano::ledger &, nano::uint128_t);
+	void observe (nano::account const &);
+	void sample ();
 	nano::uint128_t online_stake ();
-	nano::uint128_t online_stake_total;
 	std::vector<nano::account> list ();
-	boost::multi_index_container<
-	nano::rep_last_heard_info,
-	boost::multi_index::indexed_by<
-	boost::multi_index::ordered_non_unique<boost::multi_index::member<nano::rep_last_heard_info, std::chrono::steady_clock::time_point, &nano::rep_last_heard_info::last_heard>>,
-	boost::multi_index::hashed_unique<boost::multi_index::member<nano::rep_last_heard_info, nano::account, &nano::rep_last_heard_info::representative>>>>
-	reps;
+	static uint64_t constexpr weight_period = 5 * 60; // 5 minutes
+	static uint64_t constexpr weight_samples = 4032; // 2 weeks
```

Comment:
Is this comment correct?

Reply:
I think it means two weeks worth of samples - the math checks out at least. Maybe just add that to the comment.

Reply:
ah, perhaps it could be made more clear then :)

Reply:
Added a more descriptive comment